README.md

```diff
@@ -332,6 +332,62 @@ Additionally, you can also locally trigger a deep link to make sure your code is
 
 `xcrun simctl openurl booted {your_URL_here}`
 
+## Rich push notifications
+
+Rich push notification is the ability to add images, gifs and videos to your push notification messages that Apple has supported since iOS 10. In order to do this Apple requires your app to implement a [Notification service extension](https://developer.apple.com/documentation/usernotifications/unnotificationserviceextension). Following the below steps should help set up your app to receive rich push notifications.
+
+> :warning: **Note that the initial release of this feature will only support images with gifs and videos to follow.**
+
+
+### Step 1: Add notification service app extension to your project
+
+A notification service app extension ships as a separate bundle inside your iOS app. To add this extension to your app:
+
+1. Select File > New > Target in Xcode.
+2. Select the Notification Service Extension target from the iOS > Application extension section.
+3. Click Next.
+4. Specify a name and other configuration details for your app extension.
+5. Click Finish.
+
+### Step 2: Implement the notification service app extension
+
+The notification service app extension is responsible for downloading the media resource and attaching it to the push notification.
+
+Once step 1 is complete, you should see a file called `NotificationService.swift` under the notification service extension target. You can either implement it using Apple's documentation [here](https://developer.apple.com/documentation/usernotifications/modifying_content_in_newly_delivered_notifications) or use our sample code from [here](Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift).
+
+### Step 3: Test your rich push notifications
+
+#### Local testing
```

Comment:
Can it be tested from a terminal?

Reply:
Yeah I tested this on the terminal using `xcrun simctl push booted com.klaviyo.spm.example.SPMExample ~/Desktop/payload.json`  didn't work too well for me. 


Here's the payload file - 

```
{
  "aps": {
    "alert": {
      "title": "Free apple vision pro" ,
      "body": "Free Apple vision pro when you buy a Klaviyo subscription"
    },
    "mutable-content": 1
  },
  "rich-media": "https://www.apple.com/v/apple-vision-pro/a/images/overview/hero/portrait_base__bwsgtdddcl7m_large.jpg",
"rich-media-type":"jpg",
  "Simulator Target Bundle" : "com.klaviyo.spm.example.SPMExample"
}
```

Please let me know if it works for you. I might be missing something.